File: 0217-contains-duplicate/0217-contains-duplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) 
    {
      unordered_set<int> storage; //for storing numnbers from the input array
        for (int i = 0; i < nums.size(); i++) //traversing through given array of nums
        {
            if (storage.find(nums[i]) != storage.end()) // condition to check if element is already present
            {
                return true;
            }
            storage.insert(nums[i]); //insert element if no duplicates are found
        }
        
        return false;
        
    }
};

File: 0217-contains-duplicate/0217-contains-duplicate.py
class Solution(object):
    def containsDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        sizeofInput=len(nums)
        if (sizeofInput < 2):
            return False
        nums.sort()
        for i in range(sizeofInput-1):
            j = i+1
            if nums[i] == nums[j]:
                return True
        return False